lib/src/core/message/components/message_component.dart

```diff
@@ -98,17 +126,43 @@ abstract class MessageComponent implements IMessageComponent {
         return MessageMultiselect(raw);
       case 4:
         return MessageTextInput(raw);
+      case 5:
+        return MessageUserMultiSelect(raw);
+      case 6:
+        return MessageRoleMultiSelect(raw);
+      case 7:
+        return MessageMentionableMultiSelect(raw);
+      case 8:
+        return MessageChannelMultiSelect(raw);
     }
 
     throw ArgumentError("Unknown interaction type: [$type]: ${jsonEncode(raw)}");
   }
 }
 
+abstract class MultiSelectAbstract extends MessageComponent {
+  /// Custom placeholder when no option selected
+  late final String? placeholder;
+
+  /// Min value of selected options
+  late final int minValues;
+
+  /// Max value of selected options
+  late final int maxValues;
+
+  /// Whether this multiselect is disabled.
+  late final bool isDisabled;
+
+  MultiSelectAbstract(RawApiMap raw) : super(raw) {
+    placeholder = raw['placeholder'] as String?;
+    minValues = raw['min_values'] as int? ?? 1;
+    maxValues = raw['max_values'] as int? ?? 1;
+    isDisabled = raw['disabled'] as bool? ?? false;
```

Comment:
are these defaults based on something or are arbitrary?

Reply:
They're based on Discord's behaviour, whereas by default, a select menu can take only 1 to 1 (min, max) values, and are enabled.

Reply:
good